hack/install-vm.sh

```diff
@@ -141,17 +150,238 @@ EOF
     systemctl status "$ANTREA_AGENT"
 }
 
+# Generic function to execute command with timeout.
+execute_command() {
+    cmd="$1"
+    timeout="$2"
+    echo "Executing $cmd with timeout $timeout at $(date)"
+    timeout --preserve-status --foreground $timeout $cmd
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        echoerr "$cmd failed with status $ret at $(date), exiting"
+        exit 2
+    fi
+}
+
+# Install docker as container runtime.
+install_container_runtime() {
+    echo "Installing container runtime on $OS_NAME"
+    case "$OS_NAME" in
+        Ubuntu)
+        # Uninstall older versions of Docker(called docker, docker.io, or docker-engine).
+        docker_uninstall_packages="docker docker-engine docker.io containerd runc"
+        docker_install_packages="docker-ce docker-ce-cli containerd.io"
+        dependent_packages="ca-certificates curl gnupg lsb-release"
+        docker_gpg_key="/usr/share/keyrings/docker-archive-keyring.gpg"
+        docker_download_url="https://download.docker.com/linux/ubuntu"
+        apt-get remove ${docker_uninstall_packages}
+        cmd="apt-get update"
+        execute_command "$cmd" 1m
+        cmd="apt-get install -y ${dependent_packages}"
+        execute_command "$cmd" 1m
+        rm -rf ${docker_gpg_key}
+        timeout --preserve-status --foreground 10s curl -fsSL ${docker_download_url}/gpg | sudo gpg --dearmor -o ${docker_gpg_key}
+        # Setup a stable repository
+        echo \
+        "deb [arch=$(dpkg --print-architecture) signed-by=${docker_gpg_key}] ${docker_download_url} \
+        $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+        cmd="apt-get update"
+        execute_command "$cmd" 1m
+        cmd="apt-get install -y ${docker_install_packages}"
+        execute_command "$cmd" 2m
+        ;;
+    esac
+}
+
+# Install all pre-requisite packages needed for install.
+install_prerequisite_packages() {
+    set +eo pipefail
+    # Install docker, docker-compose only when running services in containers.
+    docker version
+    if [[ "$?" -ne 0 ]]; then
+        echo "Docker is not installed"
```

Comment:
This log is not required. We do have a log in install_container_runtime.

Reply:
Done.